src/components/MDXComponents/MDXLink.tsx

```diff
@@ -1,11 +1,20 @@
 import Link from 'next/link';
 import ExternalLink from '@/components/ExternalLink';
+import { useCurrentPlatform } from '@/utils/useCurrentPlatform';
 
 export const MDXLink = (props) => {
   const { href, children } = props;
   const isInternal = href && (href.startsWith('/') || href.startsWith('#'));
+
   return isInternal ? (
-    <Link href={href}>{children}</Link>
+    <Link
+      href={{
+        pathname: decodeURI(href),
```

Comment:
When we get href here the square braces around platform are actually encoded as `%5D`. Use `decodeURI` to convert the encoded square braces from `[platform]` back to the actual square braces.